WarshipGame/Assets/Scripts/Gameplay/Grid/HexData.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using UnityEngine;
+
+public enum HexType
+{
+    Default,
+    Occupied
+}
+
+/// <summary>
+/// Holds the data of every hexagon
+/// </summary>
+[SelectionBase]
+public class HexData : MonoBehaviour
+{
+    [SerializeField] private GlowManager highlight;
+    [SerializeField] private HexType hexType;
+    public Vector2Int Grid;
+
+    private void Awake()
+    {
+        highlight = GetComponent<GlowManager>();
+    }
+
+    /// <summary>
+    /// Switch Case. Hex of type Default have a value of 1, every other type is not supported
+    /// </summary>
+    /// <returns></returns>
+    /// <exception cref="Exception"></exception>
+    public int GetType()
+    {
+        switch (hexType)
+        {
+            case HexType.Default:
+                return 1;
+            default:
+                throw new Exception($"Hex of type{hexType} not supported");
+        }   
+    }
+
+    public bool IsObstacle()
+    {
+        return this.hexType == HexType.Occupied;
```

Comment:
`this.` is redundant because the program already knows it needs to grab its own `hexType`